bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableTable.java

```diff
@@ -701,6 +702,32 @@ protected boolean wasMutationApplied(
     return requestBuilder;
   }
 
+  /**
+   * For some reason, the ordering of CheckAndMutate operations is the inverse order of normal
+   * {@link ValueFilter} operations.
+   *
+   * @param compareOp
+   * @return the inverse of compareOp
+   */
+  private static CompareOp reverseCompareOp(CompareOp compareOp) {
+    switch (compareOp) {
+    case EQUAL:
+    case NOT_EQUAL:
```

Comment:
Shouldn't these be inverted also?

```
case EQUAL: 
   return CompareOp.NOT_EQUAL;
case NOT_EQUAL: 
   return CompareOp.EQUAL;
```

Doesn't look like this commit has tests for those cases but I believe the original poster encountered unexpected behavior even with strict equality?